src/drivers/heater/heater.cpp

```diff
@@ -229,9 +229,18 @@ void Heater::Run()
 			_temperature_target_met = false;
 		}
 
-		_heater_on = true;
-		heater_on();
-		ScheduleDelayed(_controller_time_on_usec);
+		if (_controller_time_on_usec > 0) {
+			// Turn the heater on.
+			_heater_on = true;
+			heater_on();
+			ScheduleDelayed(_controller_time_on_usec);
+
+		} else {
+			// Turn the heater off.
+			_heater_on = false;
+			heater_off();
+			ScheduleDelayed(_controller_period_usec);
```

Comment:
I would delete `_controller_period_usec` as a member var since it's never changed. We can just use the constant CONTROLLER_PERIOD_DEFAULT. By default this is 10ms, so the driver will check at 100Hz. Maybe could be reduced.